### PR TITLE
Add "Upgrading From Chisel 3.4 to 3.5"

### PR DESCRIPTION
**Pull request by jackkoenig**
This new document is a place to describe issues users may run into when
upgrading from Chisel 3.4 to 3.5. This initial version includes the
solution for the removal of virtual method "val io".

Inspired by the question on chisel-users: https://groups.google.com/g/chisel-users/c/OtRSISNvHeA/m/DQNzCSj3AgAJ

Here's the result of running mdoc: https://gist.github.com/jackkoenig/4949f6a455ae74923bbcce10dbf846b5

### Contributor Checklist

- [NA] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [x] Did you add appropriate documentation in `docs/src`?
- [NA] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [NA] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

   - documentation  
#### API Impact

No impact

#### Backend Code Generation Impact

No impact

#### Desired Merge Strategy

  - Squash

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (Bug fix: `3.3.x`, [small] API extension: `3.4.x`, API modification or big change: `3.5.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

File: docs/src/appendix/upgrading-from-chisel-3-4.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import chisel3._
+```
+<!-- End Prelude -->
+
+## Upgrading From Chisel 3.4 to 3.5
+
+Chisel 3.5 was a major step forward. It added support for Scala 2.13 as well as dropped many long deprecated APIs.
+Some users may run into issues while upgrading so this page serves as a central location to describe solutions to common issues.
+
+### Common Issues
```
Comment (mwachs5) [inline docs/src/appendix/upgrading-from-chisel-3-4.md:18]:
```suggestion

### General Strategy for Upgrade

Users are encouraged to first upgrade to the latest version of Chisel 3.4 (3.4.4 at the time of writing) and resolve all deprecation warnings. Doing so should enable a smoother transition to Chisel 3.5.

### Common Issues
```

**Comment by sequencer** [comment]
`remove cloneType` and force name plugin should be another two issues

**Comment by jackkoenig** [comment]
> remove cloneType and force name plugin should be another two issues 

agreed but no reason to not go ahead and merge this 🙂

**Review by seldridge — APPROVED** [review]

**Comment by jackkoenig** [comment]
Okay well that's freaky. I cancelled CI trying to make https://github.com/chipsalliance/chisel3/pull/2302 pass first and thus be merged first, but Mergify went ahead and merged the PR despite CI being cancelled 😱

**Comment by jackkoenig** [comment]
Screenshot for posterity since I am going to rerun to get rid of the red x

Screen Shot 2021-12-15 at 12 18 35 PM

**Pull request merged**